test/bot_test.exs

```diff
@@ -34,4 +35,21 @@ defmodule Slack.BotTest do
     assert slack.users    == %{"123" => %{id: "123"}}
     assert slack.ims      == %{"123" => %{id: "123"}}
   end
+
+  defmodule Slack.Rtm.Stub do
+    def start(_token) do
+      {:ok, %{url: "http://www.example.com"}}
+    end
+  end
+
+  defmodule Slack.WebsocketClient do
+    def start_link(_url, _module, _state, _options) do
+      {:ok, self()}
+    end
+  end
+
+  test "can configure the RTM module" do
+    Application.put_env(:slack, :rtm_module, Slack.Rtm.Stub)
+    assert {:ok, _pid} = :"Elixir.Slack.Bot".start_link(Bot, %{}, "token", %{client: Slack.WebsocketClient})
```

Comment:
😆 ok so after trying this I get 
```
  1) test can configure the RTM module (Slack.BotTest)
     test/bot_test.exs:51
     ** (UndefinedFunctionError) function Slack.BotTest.Slack.Bot.start_link/4 is undefined (module Slack.BotTest.Slack.Bot is not available)
     stacktrace:
       Slack.BotTest.Slack.Bot.start_link(Slack.BotTest.Bot, %{}, "token", %{client: Slack.BotTest.Slack.WebsocketClient})
       test/bot_test.exs:53: (test)
```
So thats why, but not sure why its referring to that module 😕